### PR TITLE
Gracefully exit when timeout

### PR DESCRIPTION
**Pull request by abuccts**
* Gracefully exit when timeout, add corresponding log and return code.
* Set minimum timeout to 1 minute and enlarge Ansible timeout.

### CODE REVIEW

File: superbench/runner/runner.py [hunk 1]
```diff
@@ -405,7 +408,9 @@ def _run_proc(self, benchmark_name, mode, vars):
         if mode.name == 'mpi' and mode.node_num != 1:
             ansible_runner_config = self._ansible_client.update_mpi_config(ansible_runner_config)
 
-        ansible_runner_config['timeout'] = timeout
+        if isinstance(timeout, int):
+            # we do not expect timeout in ansible unless subprocess hangs
+            ansible_runner_config['timeout'] = timeout + 300
```
Comment (cp5555) [inline superbench/runner/runner.py:413]: 300 second?

**Review by cp5555 — APPROVED** [review]

File: superbench/runner/runner.py (hunk 1 above)
Comment (guoshzhao) [inline superbench/runner/runner.py:413]: Here the timeout is for async call?

**Review by guoshzhao — APPROVED** [review]

File: superbench/runner/runner.py (hunk 1 above)
Comment (abuccts) [reply superbench/runner/runner.py:413]: yes

Comment (abuccts) [reply superbench/runner/runner.py:413]: nope, previously "timeout x command" and ansible will both kill the processes at the nearly same time, ansible timeout will not collect error/timeout logs for remote processes

now "timeout x command" will kill the processes and log related info, ansible timeout won't be triggered by default unless bash timeout hang for 300 seconds

**Pull request merged**